IOSAccessAssessment/Services/ChangesetService.swift

```diff
@@ -0,0 +1,124 @@
+        guard let url = URL(string: "\(Constants.baseUrl)/changeset/create"),
+              let accessToken
+        else { return }
+        
+        var request = URLRequest(url: url)
+        request.httpMethod = "PUT"
+        request.setValue("Bearer \(accessToken)", forHTTPHeaderField: "Authorization")
+        request.setValue(Constants.workspaceId, forHTTPHeaderField: "X-Workspace")
+        request.setValue("application/xml", forHTTPHeaderField: "Content-Type")
+        
+        guard let xmlData = """
+            <osm>
+                <changeset>
+                    <tag k="created_by" v="GIG 1.0(4)" />
+                    <tag k="comment" v="iOS OSM client" />
+                </changeset>
+            </osm>
+            """
+            .data(using: .utf8)
+        else {
+            print("Failed to create XML data.")
+            return
+        }
+
+        request.httpBody = xmlData
+        
+        URLSession.shared.dataTask(with: request) { data, response, error in
+            if let error = error {
+                completion(.failure(error))
+                return
+            }
+            
+            if let data = data, let changesetId = String(data: data, encoding: .utf8) {
+                self.changesetId = changesetId
+                completion(.success(changesetId))
+            } else {
+                completion(.failure(NSError(domain: "ChangesetError",
+                                            code: -1,
+                                            userInfo: [NSLocalizedDescriptionKey: "Failed to open changeset"])))
+            }
+        }.resume()
+    }
+    
+    func uploadChanges(nodeData: NodeData, completion: @escaping (Result<Void, Error>) -> Void) {
+        guard let changesetId,
+              let accessToken,
+              let url = URL(string: "\(Constants.baseUrl)/changeset/\(changesetId)/upload")
+        else { return }
+        
+        let xmlContent =
+        """
+        <osmChange version="0.6" generator="GIG Change generator">
+            <create>
+                <node id="-1" lat="\(nodeData.latitude)" lon="\(nodeData.longitude)" changeset="\(changesetId)" />
+            </create>
+        </osmChange>
+        """
+        
+        var request = URLRequest(url: url)
+        request.httpMethod = "POST"
+        request.setValue("Bearer \(accessToken)", forHTTPHeaderField: "Authorization")
+        request.setValue(Constants.workspaceId, forHTTPHeaderField: "X-Workspace")
+        request.setValue("application/xml", forHTTPHeaderField: "Content-Type")
+        request.httpBody = xmlContent.data(using: .utf8)
+        
+        URLSession.shared.dataTask(with: request) { data, response, error in
+            if let error = error {
+                completion(.failure(error))
+                return
+            }
+            completion(.success(()))
+        }.resume()
+    }
+    
+    func closeChangeset(completion: @escaping (Result<Void, Error>) -> Void) {
```

Comment:
I wonder if we should be closing the ChangeSet in the ContentView as well, every time the back button is clicked.

Reply:
Nevermind. Change sets are generally closed automatically when this happens